### ISSUE TITLE
Does not distinguish between templates and instance units

### ISSUE THREAD

**Issue by shadoi** [CenturyLinkLabs/fleet-api#6]
If I load a unit as a template using the format name@.service, fleet-api creates a job for it, which fleet then loops and complains about on whatever machine it scheduled to.  If a unit is identified as a template it should not create a job.  (I'm assuming this is the extent of the problem)

Launching instances of this template fails after this (jobs are created but nothing happens and they don't show up in the output of `fleetctl list-units`

This happens on both 0.8.3 and 0.9.0

log output:
```plaintext
ERROR generator.go:51: Failed fetching current unit states: Unit name name@.service is not valid.
```

**Comment by bdehamer** [comment]
From my investigation, it looks like the template stuff is really a feature of the `fleetctl` client and not the API itself. If you do something like `fleetctl start foo@1.service` it results in API traffic that looks like this:

    GET /fleet/v1/units/foo@1.service // which results in a 404
    GET /fleet/v1/units/foo@service 
    PUT /fleet/v1/units/foo@1.service
    GET /fleet/v1/units/foo@1.service // 200, this time

Basically, `fleetctl` will pull down the template and then turn around and submit it as a new unit with the specified name.

Would welcome a PR if someone wants to take a crack at implementing something like this.

**Comment by robholland** [comment]
Recent fleet versions now prevent you from trying to start a template unit. We had so much pain with this, so easy to accidentally do it. I think you can close this, fleet will do the right thing (return an error).

**Comment by bdehamer** [comment]
@robholland, thanks for the update. Gonna close this issue.
